pkg/gencosmosdb/cosmosdb/cosmosdb.go

```diff
@@ -121,12 +127,21 @@ func (c *databaseClient) do(ctx context.Context, method, path, resourceType, res
 			d.Decode(&err)
 		}
 		err.StatusCode = resp.StatusCode
-		return err
+		return resp, err
+	}
+
+	if headers != nil {
```

Comment:
moving this was a hack - needs to be dealt with by separating headers to something like "inheaders" and "outheaders" to avoid the overwrite

Reply:
In looking at the current code and a little bit confused, why change is needed in the first place? 
```
	if headers != nil {
		for k := range headers {
			delete(headers, k)
		}
		for k, v := range resp.Header {
			headers[k] = v
		}
	}
```
Can we just remove it and replace it with `inHeader := header` at the top to copy values to make sure we play with a local copy?